relay.go

```diff
@@ -85,6 +87,11 @@ func NewRelay(ctx context.Context, h host.Host, upgrader *tptu.Upgrader, opts ..
 		incoming: make(chan *Conn),
 		relays:   make(map[peer.ID]struct{}),
 		liveHops: make(map[peer.ID]map[peer.ID]int),
+		bufPool: sync.Pool{
+			New: func() interface{} {
+				return make([]byte, 4096)
```

Comment:
So, welcome to go. Converting a `[]byte` to an `interface{}` _allocates_. Let's just use https://github.com/libp2p/go-buffer-pool (https://godoc.org/github.com/libp2p/go-buffer-pool#Get)

Reply:
Oh wow, that's nuts.
Yeah, I'll use go-buffer-pool.